conans/model/scm.py

```diff
@@ -27,6 +34,7 @@ def __init__(self, conanfile):
             self.password = data.get("password")
             self.subfolder = data.get("subfolder")
             self.submodule = data.get("submodule")
+            self._shallow = SCMData._boolean(data.get("shallow"), None)
```

Comment:
I think the internal value of shallow should already be a valid boolean, and not None, once defined. So avoid the below ``return self._shallow in [None, "True"]`` which is confusing.

Reply:
I just wanted to leave existing behavior unmodified. If I assign a True/False value then it will be printed to the `scm` attribute in the `conanfile.py` and we won't be able to differentiate if the user has explicitly set a value or not... a collateral effect will be that the same `conanfile.py` will have a different hash (revision) with this new Conan v1.18.2.

Of course, I can remove the attibute if the value is `True` (default value) and avoid printing it to the recipe.

Reply:
> Of course, I can remove the attibute if the value is True (default value) and avoid printing it to the recipe.

I think that might be better.